pkg/system/phase2_creating.go

```diff
@@ -31,41 +31,57 @@ func (r *Reconciler) ReconcilePhaseCreating() error {
 		"noobaa operator started phase 2/4 - \"Creating\"",
 	)
 
-	// A failure to discover OAuth endpoints should not fail the entire reconcile phase.
-	oAuthEndpoints, err := util.DiscoverOAuthEndpoints()
-	if err != nil {
-		r.Logger.Warnf("Discovery of OAuth endpoints failed, got: %v", err)
+	if r.JoinSecret == nil {
```

Comment:
Isn't it better to split into two ReconcilePhaseCreating? One for Join nil and for for !nil?

Reply:
sounds better to me too

Reply:
@nimrod-becker @guymguym 
I don't think so. It creates duplicate code that needs to be updated for every change in both places.
Also, the end goal (in the future) is to have full NooBaa on the remote cluster which means 
that we will take reconcile function out of the if statements little by little

Reply:
at the moment it's not very readable. there is no need to duplicate code. if there is a lot of common code you can extract it to a function

Reply:
lets not create duplicate code. Create a common base which runs in both cases and then a specific func for each case.

Reply:
I changed the way the code is structured. The main change is that each reconcile function checks for itself if it needs to be skipped